pkg/gatherers/conditional/gathering_rule.schema.json

```diff
@@ -0,0 +1,127 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "gathering_rule.schema.json",
+    "title": "Conditional Gatherer Rule",
+    "descriptions": "A gathering rule for conditional gatherer",
+    "examples": [
+        {
+            "conditions": [
+                {
+                    "type": "alert_is_firing",
+                    "params": {
+                        "name": "SamplesImagestreamImportFailing"
+                    }
+                }
+            ],
+            "gathering_functions": {
+                "gather_logs_of_namespace": {
+                    "namespace": "openshift-cluster-samples-operator",
+                    "keep_lines": 100,
+                    "label_selector": ""
+                },
+                "gather_imagestreams_of_namespace": {
+                    "namespace": "openshift-cluster-samples-operator"
+                }
+            }
+        }
+    ],
+    "type": "object",
+    "required": [
+        "conditions",
+        "gathering_functions"
+    ],
+    "properties": {
+        "conditions": {
+            "type": "array",
+            "description": "The list of conditions on which the rule is triggered",
+            "minItems": 0,
+            "maxItems": 8,
+            "uniqueItems": true,
+            "items": {
+                "type": "object",
+                "title": "ConditionWithParams",
+                "description": "A condition",
+                "anyOf": [
+                    {
+                        "type": "object",
+                        "title": "ConditionWithParams",
+                        "description": "alert_is_firing condition",
+                        "required": [
+                            "type",
+                            "params"
+                        ],
+                        "properties": {
+                            "type": {
+                                "type": "string",
+                                "title": "Type",
+                                "description": "Type of the condition alert_is_firing",
+                                "const": "alert_is_firing"
+                            },
+                            "params": {
```

Comment:
Hmm now when I am thinking about this, it could be a bit confusing. `params` word sounds more like an array of params to me (which is not the case if I understand it correctly).

Do we allow more alert names (within one condition)? I guess there could be such condition in the future. How would it be defined?

Reply:
It's called params because for other type of alert it can be different. We can define a rule requiring 2 alerts to fire like this:
```{
  "conditions": [
    {
      "type": "alert_is_firing",
      "params": {
        "name": "SamplesImagestreamImportFailing"
      }
    },
    {
      "type": "alert_is_firing",
      "params": {
        "name": "AnotherAlert"
      }
    }
  ],
  "gathering_functions": {
    "logs_of_namespace": {
      "namespace": "openshift-cluster-samples-operator",
      "tail_lines": 100
    },
    "image_streams_of_namespace": {
      "namespace": "openshift-cluster-samples-operator"
    }
  }
}```

Reply:
OK thanks. Yes. It's probably just a detail, but maybe we could think about changing the `params` attribute to array..... I guess it would still have to be an array of objects.

Reply:
Why? Map fits better in my opinion, cuz we're not gonna have multiple names for an alert for example.

Reply:
I was thinking mainly of the size of the JSON file TBH. So something like:
```
"conditions": [
    {
      "type": "alert_is_firing",
      "params": [
        "name": "AlertA",
        "name": "AlertB", 
      ]
    },
  ],
```
This could be problematic from your point of view (why? validation?)?

Reply:
Well, we can have something like:
```
  "conditions": [
    {
      "type": "alerts_are_firing",
      "params": [
        { "name": "AlertA" },
        { "name": "AlertB" }
      ]
    }
  ],
```
But we'd need to change the operator's code, the schema (it may become more complicated) and so on. The problem can be for some conditions which can't have multiple parameters of the same type, we'd need to decide on the behaviour when there's multiple params of the same type (can be use the latest, return an error, etc).

Reply:
I see thanks. Good points. We don't have to address it right now, but I think we should keep that in mind (maybe new JIRA task?), because this requirement (having a condition with more alerts) is very likely to come IMO.